src/tokenized-buffer.coffee

```diff
@@ -349,13 +349,23 @@ class TokenizedBuffer extends Model
 
   indentLevelForLine: (line) ->
     if match = line.match(/^[\t ]+/)
-      leadingWhitespace = match[0]
-      tabCount = leadingWhitespace.match(/\t/g)?.length ? 0
-      spaceCount = leadingWhitespace.match(/[ ]/g)?.length ? 0
-      tabCount + (spaceCount / @getTabLength())
```

Comment:
Yes it was surprising! I hope my method is better.

It also follows tab stops if there's spaces intermingled. I know mixed tab/space is REALLY bad anyway but it hopefully means indentLevelForLine returns nice correct numbers that match exactly the visual indentation we see, where as at moment it would be weird. For example "S\tS\t" would return 3 where visually it would actually be identical to "\t\t" which is 2 (assuming 2 space tab)